app/controllers/api/v2/base_controller.rb

```diff
@@ -64,6 +66,10 @@ def metadata_total
         @total ||= resource_scope.try(:count).to_i
       end
 
+      def can_create?
+        authorized_for({:controller => controller_permission, :action => "create", :user_id => User.current})
```

Comment:
Nit: this can be written as `authorized_for(controller: controller_permission, action: 'create', user_id: User.current)`, which is more modern ruby style.